### PR TITLE
yogisha/DELETE functionalities for cart

### PR DESCRIPTION
**Pull request by YogishaNiraula**

### CODE REVIEW

File: src/app/[affiliateSlug]/cart/CartPage.tsx [hunk 1]
```diff
@@ -6,18 +6,37 @@ import { CartItem, Carts } from "@/types/cart";
...
+      const updatedCart = await response.json();
+      setCart(updatedCart.cart_items);
+      setSubtotal(updatedCart.sub_total);
+    } catch (error) {
+      console.error("Error fetching cart:", error);
+    }
+  };
+
+  useEffect(() => {
+    refetchCart();
```
Comment (nardbagel) [inline src/app/[affiliateSlug]/cart/CartPage.tsx:38]: this is to sync the state if the widget adds more to the cart (by pressing +)?

File: src/app/[affiliateSlug]/cart/CartPage.tsx [hunk 2]
```diff
@@ -6,18 +6,37 @@ import { CartItem, Carts } from "@/types/cart";
...
-  const [cart, setCart] = useState<CartItem[]>(cartInfo?.cart_items ?? []);
+  const [cart, setCart] = useState<CartItem[] | []>(cartInfo?.cart_items ?? []);
   const [subtotal, setSubtotal] = useState<string>(
     cartInfo?.sub_total ?? "0.00"
   );
+  const [cookie, setCookie] = useCookies();
+
+  const refetchCart = async () => {
+    try {
+      const response = await fetch("/api/cart");
```
Comment (nardbagel) [inline src/app/[affiliateSlug]/cart/CartPage.tsx:26]: can we refactor this into its own api function with a zod parser?

File: src/app/[affiliateSlug]/cart/CartPage.tsx [hunk 3]
```diff
@@ -32,39 +51,56 @@ export default function CartPage({ cartInfo, slug }: CartPageProps) {
...
+  const handleQuantityChange = (id: number, quantity: number) => {
+    // setCart((prevCart) =>
+    //   prevCart.map((item) => (item.id === id ? { ...item, quantity } : item))
+    // );
+    updateQuantity(id, quantity);
+  };
+
+  const handleItemDelete = async (id: number) => {
+    try {
+      const response = await fetch("/api/cart", {
```
Comment (nardbagel) [inline src/app/[affiliateSlug]/cart/CartPage.tsx:87]: same here. Lets move this into its own file in /api and do a zod parse on it.

**Review by nardbagel — CHANGES_REQUESTED** [review]
I think we just move the api endpoint calls to their own functions in the /api directory, and use zod parse like we are doing with the other API calls. Other than that, looks great!

File: src/app/[affiliateSlug]/cart/CartPage.tsx (hunk 1 above)
Comment (YogishaNiraula) [reply src/app/[affiliateSlug]/cart/CartPage.tsx:38]: Yes, its for quantity changes sync

**Review by nardbagel — APPROVED** [review]

**Pull request merged**